docs/next/guides/building-and-testing/modules.md

```diff
@@ -481,352 +275,822 @@ new Handsontable(container, {});
+
+import { registerLanguageDictionary } from 'handsontable/i18n/registry';
+```
+:::
+
+## Using modules
+
+To get the most out of using Handsontable modules:
+1. [Import the required modules](#importing-required-modules).
+2. Use optional modules of your choice:
+   - [Editor modules](#using-editor-modules)
+   - [Renderer modules](#using-renderer-modules)
+   - [Validator modules](#using-validator-modules)
+   - [Cell type modules](#using-cell-type-modules)
+   - [Plugin modules](#using-plugin-modules)
+   - [Translation modules](#using-translation-modules)
+3. Remove redundant code, using [tree shaking](#tree-shaking).
+
+### Importing required modules
+
+No matter which of the [optional modules](#list-of-all-modules) you use, the following modules are always required:
+
+- Handsontable's core functionalities
+- The default [cell type](@/guides/cell-types/cell-type.md): `text`
+- [moment.js](https://momentjs.com/)
+- [DOMpurify](https://www.npmjs.com/package/dompurify)
+- [core-js](https://www.npmjs.com/package/core-js)
+- Handsontable's CSS
+
+To import the required modules, follow the steps below.
+
+#### Step 1: Remove the full Handsontable import
+
+When using modules, you no longer need to import the full version of Handsontable.
+
+Remove the `handsontable` import:
+
+```js
+// remove this line
+import Handsontable from 'handsontable';
+```
+
+#### Step 2: Import the `handsontable/base` module
+
+Instead of the full Handsontable version, import the `handsontable/base` module.
 
-To use a specific editor you need to import it alongside the registering method. For instance, let's try adding the password editor.
+```js
+// import the `handsontable/base` module
+import Handsontable from 'handsontable/base';
+```
+
+#### Step 3: Import the CSS
+
+Handsontable's CSS file is not modularized, so you need to import it:
+
+```js
+import Handsontable from 'handsontable/base';
+
+// import Handsontable's CSS
+import 'handsontable/dist/handsontable.full.css';
+```
+
+Now, you're ready to use [any optional modules](#list-of-all-modules) of your choice.
+
+### Using editor modules
+
+Each editor module contains a different type of [cell editor](@/guides/cell-functions/cell-editor.md):
 
-Start with importing the base, `PasswordEditor` and the `registerEditor` method.
+| Module               | Cell editor alias |
+| -------------------- | ----------------- |
+| `AutocompleteEditor` | `autocomplete`    |
+| `BaseEditor`         | `base`            |
+| `CheckboxEditor`     | `checkbox`        |
+| `DateEditor`         | `date`            |
+| `DropdownEditor`     | `dropdown`        |
+| `HandsontableEditor` | `handsontable`    |
+| `NumericEditor`      | `numeric`         |
+| `PasswordEditor`     | `password`        |
+| `SelectEditor`       | `select`          |
+| `TextEditor`         | `text`            |
+
+To start using an editor module, follow the steps below.
+
+#### Step 1: Import the required modules
```

Comment:
I don't think that explaining this in steps is useful. Can you convince me? I just think that it takes twice the amount of screen space and adds unnecessary repetition.

This is the same remark as on line 12 of `vue-modules.md`.

Reply:
TL;DR
- The purpose of a "guide" is to take the user through a standard procedure that results in a specific outcome. And procedures consist of numbered steps.
- Additionally, numbered steps suggest a standard way of using a feature.
- Additionally, it's better to underestimate your users' background knowledge than overestimate it.

---

Here's where I'm coming from:

First, what I'm trying to do here is apply some content management rules. In a content management methodology (be it [topic-based writing](https://en.wikipedia.org/wiki/Topic-based_authoring) or [modular writing](https://writing.stackexchange.com/questions/37891/topic-based-authoring-vs-modular-authoring)), the first thing is to determine elementary content types. A classic example is [DITA](https://en.wikipedia.org/wiki/Darwin_Information_Typing_Architecture), where some of the basic content types are:
- [Reference topic](https://docs.oasis-open.org/dita/v1.2/os/spec/archSpec/dita_reference_topic.html)
- [Concept topic](https://docs.oasis-open.org/dita/dita/v1.3/os/part2-tech-content/archSpec/technicalContent/dita-concept-topic.html)
- [Task topic](https://heretto.com/how-to-write-a-good-dita-task-topic/)

Don't get me wrong: I'm not trying to apply DITA rules here - it's just that docs are not a blog meant to be read: they are independent, interlinked pieces of content that are meant to be:
- Quickly looked up (reference)
- Scanned for a broader understanding (concept)
- Followed,  to achieve a specific outcome (tasks, or "guides" if you will).

Since our documentation is not based on any specific methodology, I'm trying to at least apply a standard structure that properly separates different content types that happen to be (for better or worse) on the same page, for example:
- Section (with a unique name)
  - Page (with a unique title)
    - Short introduction (max 1 sentence)
    - Necessary conceptual information
    - Tasks/guides/procedures:
      - Achieving an outcome no. 1
        - Step 1
        - Step 2
        - Step 3
        - Outcome
      - Achieving an outcome no. 2
        - Step 1
        - Step 2
        - Step 3
        - Outcome

If an aim of a piece of information is to make the reader achieve a certain outcome, it's a task. Tasks cover procedures. Procedures, by definition, are made of numbered steps, to ensure the proper outcome and limit the number of possible applications to just one that we can easily test. And finally: each step covers just one action (ideally).

Hence the need to structure the content in a predictible, replicable (ideally: reusable - e.g. in a [CCMS](https://en.wikipedia.org/wiki/Component_content_management_system), each "duplicate" step would be a reusable snippet) manner, which is what I'm trying to enforce here.

Presenting a [loosely arranged group of headings](https://gist.github.com/warpech/f9a712ce507708c5eeb6ccdfb97d4609) goes the opposite direction.

Additionally, I treat editor modules as a feature, and numbered steps suggest a standard way of using such feature. And if we **can** suggest a standard way of using a feature, IMO we should, because:
- It provides guidance to inexperienced developers
- It suggests the best way to use a feature (e.g. performance-wise)
- A specific scenario is easier to test than a general scenario or no scenario at all

Additionally, it's better to underestimate your users' background knowledge than overestimate it. IMO our documentation should:
- Underestimate the user's command of English
- Underestimate the user's command of JavaScript (and/or frameworks)
- Underestimate the user's familiarity with Handsontable
- Underestimate the user's familiarity with any of the other Handsontable documentation pages (it's called "[every page is page one](https://www.stc.org/notebook/2013/10/01/every-page-is-page-one/)" rule)

Of course, this is all debatable and ultimately up to you :)

Reply:
Thanks for the context. You take the word "guide" quite literally (like a tourist guide). I take it as the following [definition](https://dictionary.cambridge.org/dictionary/english/guide):

> a book that gives you the most important information about a particular subject

On the general "guide" pages, we just explain topics in higher detail than on the API reference pages, showing some context behind that. We show the reader his options, but we don't know his goal, so it is hard to throw "steps" at this stage. We need to present content in a way that the steps will be apparent for the reader.

If we ever add "how-to guides", these would be pages that explain how to add a certain tasks.

We are not the only ones with vision for a guide:

- https://ckeditor.com/docs/ckeditor4/latest/guide/index.html (section called Guide). Notably it is not called Guide in version 5 (just "Docs"): https://ckeditor.com/docs/ckeditor5/latest/features/index.html
- https://reactjs.org/docs/accessibility.html (section Advanced Guides)
- https://v3.vuejs.org/guide/introduction.html (section Guide). I find this particularly good example of what I mean.